xclim/indicators/land/_streamflow.py

```diff
@@ -79,7 +79,7 @@ def cfcheck(self, **das):
     "distribution.",
     abstract="Streamflow frequency analysis on the basis of a given mode and distribution.",
     units="m^3 s-1",
-    compute=declare_units(da=None)(frequency_analysis),
+    compute=declare_units(da="[discharge]")(frequency_analysis),
```

Comment:
```suggestion
    compute=frequency_analysis,
    inputs={'da': 'q'}
```
This is my up-sleeve trick! It sets the correct dimensionality (and units check), expected standard name and expected cell_methods.

The trick is currently limited by #1129 as the new name (`q`) must be documented in `data/variables.yml`. And it is currently not there.

Originally meant to allow changing the variable name of a virtual-module indicator to make the use of `ind(ds=ds)` work, we realized this technique could be useful to wrap generic indices in a meaningful but easy way.  The code that manages it is here : https://github.com/Ouranosinc/xclim/blob/d565f5c0985a3eda09c395b0b7dfc4ffdb764393/xclim/core/indicator.py#L563

Reply:
Ok, so I understand we're OK with breaking backward compatibility in an effort to have consistent input variable naming patterns across indicators?

Are we good with the name `discharge`?

Reply:
The decision for backwards compatibility is more so in your hands (aka, the xclim group), but `discharge` sounds good to me.

Reply:
If it's possible to keep both (ie: their names are different), I'm a big fan of indicating that an indicator will be deprecated. There's a special field in `Indicator` classes for marking an indicator as being deprecated: https://github.com/Ouranosinc/xclim/blob/d565f5c0985a3eda09c395b0b7dfc4ffdb764393/xclim/core/indicator.py#L360

Reply:
It's the same indicator, but with a different signature, so I don't think it applies here.

Reply:
I am ok with backward compatibility for this small set of indicators.
On the other hand, now that the indicators are not _generic_ anymore, maybe the name should reflect that?
Ex:
`freq_analysis` -> `discharge_return_period_amount`
`fit` -> `discharge_distrtibution`
?
The three others already have discharge-related names. Quoique, `doy_qmax` is a bit confusing now that we are using the `discharge` variable name.

Reply:
I agree that `freq_analysis/fit` should probably be generic (for use with something like precipitation, too), while discharge-specific indicators should have a more explicit name.